src/backend/distributed/planner/shard_pruning.c

```diff
@@ -1614,20 +1630,30 @@ LowerShardBoundary(Datum partitionColumnValue, ShardInterval **shardIntervalCach
 	 * (we'd have hit the return middleIndex; case otherwise). Figure out
 	 * whether there's possibly any interval containing a value that's bigger
 	 * than the partition key one.
+	 *
+	 * Also note that we initialized lowerBoundIndex with 0. Similarly,
+	 * we always set it to the index of the  shard that we consider as our
+	 * lower boundary during binary search.
 	 */
-	if (lowerBoundIndex == 0)
-	{
-		/* all intervals are bigger, thus return 0 */
-		return 0;
-	}
-	else if (lowerBoundIndex == shardCount)
+	if (lowerBoundIndex == shardCount)
 	{
-		/* partition value is bigger than all partition values */
+		/*
+		 * Since lowerBoundIndex is an inclusive index, INVALID_SHARD_INDEX
+		 * means all the shards have smaller values than partitionColumnValue,
```

Comment:
```suggestion
		 * Since lowerBoundIndex is an inclusive index, if it equal to shardCount then that
		 * means all the shards have smaller values than partitionColumnValue,
```